src/compiler/scala/tools/nsc/transform/SpecializeTypes.scala

```diff
@@ -608,15 +609,16 @@ abstract class SpecializeTypes extends InfoTransform with TypingTransformers {
       bytecodeClazz.info
 
       val sClass = clazz.owner.newClass(clazzName, clazz.pos, (clazz.flags | SPECIALIZED) & ~CASE)
-      sClass.setAnnotations(clazz.annotations) // scala/bug#8574 important that the subclass picks up @SerialVersionUID, @strictfp, etc.
+        .tap(_.setAnnotations(clazz.annotations))
 
       def cloneInSpecializedClass(member: Symbol, flagFn: Long => Long, newName: Name = null) =
         member.cloneSymbol(sClass, flagFn(member.flags | SPECIALIZED), newName)
 
-      val sourceFile = clazz.sourceFile
-      if (sourceFile ne null) {
-        sClass.associatedFile = sourceFile
-        currentRun.symSource(sClass) = sourceFile // needed later on by mixin
+      val sourceFile = clazz.sourceFile.tap { sf =>
+        if (sf ne null) {
+          sClass.associatedFile = sf
+          currentRun.symSource(sClass) = sf // needed later on by mixin
+        }
```

Comment:
to me, using `tap` here doesn't improve readability

Reply:
I was getting tapped out. If I'd remembered to mark this a draft PR, I could have punned on draft beer and tap. I will undo the tap dance.

Reply:
The pertinent question was, "Is sourceFile even used?" Now that lint is on, we know the answer. It was only used as an alias in the if block.

My other second thought is that I now prefer `val x = v.tap(init)` to mean assign v to x, but do some side-effecting set-up first.